### ISSUE TITLE
`npx create-remix@latest` says `Oh no! Failed to initialize git.`

### ISSUE THREAD

**Issue by hytdsh** [remix-run/remix#9057]
### Reproduction

```
node@c2ea7dfbb793:~$ node -v
v20.11.1
node@c2ea7dfbb793:~$ npm -v
10.2.4
```

```
node@c2ea7dfbb793:~$ npx create-remix@latest
Need to install the following packages:
create-remix@2.8.1
Ok to proceed? (y)

 remix   v2.8.1 💿 Let's build a better website...

   dir   Where should we create your new project?
         ./some-app

      ◼  Using basic template See https://remix.run/guides/templates for more
      ✔  Template copied

   git   Initialize a new git repository?
         Yes

  deps   Install dependencies with npm?
         Yes

      ✔  Dependencies installed

 ██████  Git initializing...

      ▲  Oh no! Failed to initialize git.
npm notice
npm notice New minor version of npm available! 10.2.4 -> 10.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.0
npm notice Run npm install -g npm@10.5.0 to update!
npm notice
```

```
node@c2ea7dfbb793:~$ ls
some-app
node@c2ea7dfbb793:~$ cd some-app
node@c2ea7dfbb793:~/some-app$ ls -la
total 460
drwxr-xr-x   6 node node   4096 Mar 15 07:20 .
drwxr-xr-x   1 node node   4096 Mar 15 07:19 ..
-rw-r--r--   1 node node   1837 Mar 15 07:19 .eslintrc.cjs
drwxr-xr-x   7 node node   4096 Mar 15 07:20 .git
-rw-r--r--   1 node node     48 Mar 15 07:19 .gitignore
-rw-r--r--   1 node node    578 Mar 15 07:19 README.md
drwxr-xr-x   3 node node   4096 Mar 15 07:19 app
drwxr-xr-x 597 node node  20480 Mar 15 07:20 node_modules
-rw-r--r--   1 node node 397945 Mar 15 07:20 package-lock.json
-rw-r--r--   1 node node   1104 Mar 15 07:19 package.json
drwxr-xr-x   2 node node   4096 Mar 15 07:19 public
-rw-r--r--   1 node node    242 Mar 15 07:19 remix.config.js
-rw-r--r--   1 node node     83 Mar 15 07:19 remix.env.d.ts
-rw-r--r--   1 node node    570 Mar 15 07:19 tsconfig.json
node@c2ea7dfbb793:~/some-app$ git status
On branch master

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
        new file:   .eslintrc.cjs
        new file:   .gitignore
        new file:   README.md
        new file:   app/entry.client.tsx
        new file:   app/entry.server.tsx
        new file:   app/root.tsx
        new file:   app/routes/_index.tsx
        new file:   package-lock.json
        new file:   package.json
        new file:   public/favicon.ico
        new file:   remix.config.js
        new file:   remix.env.d.ts
        new file:   tsconfig.json

node@c2ea7dfbb793:~/some-app$
```

### System Info

```shell
node@c2ea7dfbb793:~$ npx envinfo --system --npmPackages '{vite,@remix-run/*}' --binaries --browsers
Need to install the following packages:
envinfo@7.11.1
Ok to proceed? (y)

  System:
    OS: Linux 5.15 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Memory: 14.27 GB / 15.62 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 20.11.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.4 - /usr/local/bin/npm

node@c2ea7dfbb793:~$
```
```


### Used Package Manager

npm

### Expected Behavior

`npx create-remix@latest` completes with no error/warning.

### Actual Behavior

`npx create-remix@latest` says `Oh no! Failed to initialize git.`

**Comment by brophdawg11** [comment]
It looks like maybe the underlying git error isn't being surfaced properly here.  Can you try running `git commit -m "test"` inside the created `some-app/` folder and let us know what the error is?  That should give us a way to reproduce and we can look into adding the error test to the output.

**Comment by hytdsh** [comment]
Thank you for your comment.
Here is the log.

compose.yml
```
version: '3'

services:
  app:
    build:
      context: ./app
      dockerfile: Dockerfile
    ports:
      - 3000:3000
    environment:
      - NODE_ENV=development
```

app/Dockerfile
```
FROM node:20.11.1-bookworm

RUN apt-get update && apt-get install -y \
        curl \
        git \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

USER node
WORKDIR /home/node

CMD ["tail", "-f", "/dev/null"]
```

log
```
~/work/etude$ docker compose up -d --build
[+] Building 0.9s (7/7) FINISHED                                                        docker:default
 => [app internal] load build definition from Dockerfile                                          0.0s
 => => transferring dockerfile: 400B                                                              0.0s
 => [app internal] load metadata for docker.io/library/node:20.11.1-bookworm                      0.7s
 => [app internal] load .dockerignore                                                             0.0s
 => => transferring context: 2B                                                                   0.0s
 => [app 1/3] FROM docker.io/library/node:20.11.1-bookworm@sha256:e06aae17c40c7a6b5296ca6f942a02  0.0s
 => CACHED [app 2/3] RUN apt-get update && apt-get install -y         curl         git     && ap  0.0s
 => [app 3/3] WORKDIR /home/node                                                                  0.0s
 => [app] exporting to image                                                                      0.0s
 => => exporting layers                                                                           0.0s
 => => writing image sha256:005b640767480d00afa357a141c380b30fcd5fab45b48e03270aadcb0496a02b      0.0s
 => => naming to docker.io/library/etude-app                                                      0.0s
[+] Running 1/2
 ⠴ Network etude_default  Created                                                                 0.4s
 ✔ Container etude-app-1  Started                                                                 0.4s

~/work/etude$ docker compose exec app bash
node@0ea8e3e26654:~$ pwd
/home/node
node@0ea8e3e26654:~$ node -v
v20.11.1
node@0ea8e3e26654:~$ npm -v
10.2.4

node@0ea8e3e26654:~$ npx create-remix@latest
Need to install the following packages:
create-remix@2.8.1
Ok to proceed? (y)

 remix   v2.8.1 💿 Let's build a better website...

   dir   Where should we create your new project?
         ./some-app

      ◼  Using basic template See https://remix.run/guides/templates for more
      ✔  Template copied

   git   Initialize a new git repository?
         Yes

  deps   Install dependencies with npm?
         Yes

      ✔  Dependencies installed


      ▲  Oh no! Failed to initialize git.
npm notice
npm notice New minor version of npm available! 10.2.4 -> 10.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.0
npm notice Run npm install -g npm@10.5.0 to update!
npm notice

node@0ea8e3e26654:~$ cd some-app
node@0ea8e3e26654:~/some-app$ git status
On branch master

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
        new file:   .eslintrc.cjs
        new file:   .gitignore
        new file:   README.md
        new file:   app/entry.client.tsx
        new file:   app/entry.server.tsx
        new file:   app/root.tsx
        new file:   app/routes/_index.tsx
        new file:   package-lock.json
        new file:   package.json
        new file:   public/favicon.ico
        new file:   remix.config.js
        new file:   remix.env.d.ts
        new file:   tsconfig.json

node@0ea8e3e26654:~/some-app$ git config --global user.email "hytdsh@example.com"
node@0ea8e3e26654:~/some-app$ git config --global user.name "hytdsh"

node@0ea8e3e26654:~/some-app$ git commit -m "test"
[master (root-commit) c939e5c] test
 13 files changed, 11383 insertions(+)
 create mode 100644 .eslintrc.cjs
 create mode 100644 .gitignore
 create mode 100644 README.md
 create mode 100644 app/entry.client.tsx
 create mode 100644 app/entry.server.tsx
 create mode 100644 app/root.tsx
 create mode 100644 app/routes/_index.tsx
 create mode 100644 package-lock.json
 create mode 100644 package.json
 create mode 100644 public/favicon.ico
 create mode 100644 remix.config.js
 create mode 100644 remix.env.d.ts
 create mode 100644 tsconfig.json

node@0ea8e3e26654:~/some-app$ git log
commit c939e5c50be86e063dd3d670c46a5e20045d26b0 (HEAD -> master)
Author: hytdsh <hytdsh@example.com>
Date:   Tue Mar 19 07:59:34 2024 +0000

    test
```

**Comment by brophdawg11** [comment]
You did more than just run `git commit -m "..."` there after the failure 😉 

You added the commands to set `user.name` and `user.email` so my hunch is that's what's failing internally.  Are those not properly set prior to running `create-remix`?  If not, I would set them - otherwise I would just choose `no` to the Git question and then manually cd into the directory afterwards, set those up and run `git init` yourself.

I'll see about getting the underlying error better exposed

**Comment by hytdsh** [comment]
I see! Thanks!

I was able to run it without error by `git config` before `npx create-remix@latest`.

```
~/work/etude$ docker compose exec app bash
node@0efd5ea9cc6f:~$
node@0efd5ea9cc6f:~$ git config --global user.email "hytdsh@example.com"
node@0efd5ea9cc6f:~$ git config --global user.name "hytdsh"
node@0efd5ea9cc6f:~$
node@0efd5ea9cc6f:~$ git config --list
user.email=hytdsh@example.com
user.name=hytdsh

node@0efd5ea9cc6f:~$ npx create-remix@latest
Need to install the following packages:
create-remix@2.8.1
Ok to proceed? (y)

 remix   v2.8.1 💿 Let's build a better website...

   dir   Where should we create your new project?
         ./some-app

      ◼  Using basic template See https://remix.run/guides/templates for more
      ✔  Template copied

   git   Initialize a new git repository?
         Yes

  deps   Install dependencies with npm?
         Yes

      ✔  Dependencies installed

      ✔  Git initialized

  done   That's it!

         Enter your project directory using cd ./some-app
         Check out README.md for development and deploy instructions.

         Join the community at https://rmx.as/discord

npm notice
npm notice New minor version of npm available! 10.2.4 -> 10.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.0
npm notice Run npm install -g npm@10.5.0 to update!
npm notice
```

**Comment by brophdawg11** [comment]
Oh, sorry I forgot we do have a way to surface those errors - you can add the `--debug` flag to get the underlying error logged 👍

**Comment by hytdsh** [comment]
Here is the log with --debug.
I was thinking, perhaps it's not necessary to go as far as doing a git commit. Wouldn't just doing git init be enough?

```
~/work/etude$ docker compose exec app bash
node@123222c0b4fa:~$ npx create-remix@latest --debug
Need to install the following packages:
create-remix@2.8.1
Ok to proceed? (y)

 remix   v2.8.1 💿 Let's build a better website...

   dir   Where should we create your new project?
         ./some-app

      ◼  Using basic template See https://remix.run/guides/templates for more
      ●  Extracting to: /tmp/create-remix--qx2uqltg
      ●  Using the template from "https://github.com/remix-run/remix/tree/main/templates/remix"
      ✔  Template copied

   git   Initialize a new git repository?
         Yes

  deps   Install dependencies with npm?
         Yes

      ✔  Dependencies installed


      ▲  Oh no! Failed to initialize git.
Error: Command failed with exit code 128: git commit -m Initial commit from create-remix
    at makeError (/home/node/.npm/_npx/5164864a48bff686/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/home/node/.npm/_npx/5164864a48bff686/node_modules/execa/index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async while (/home/node/.npm/_npx/5164864a48bff686/node_modules/create-remix/dist/index.js:424:9)
    at async Object.renderLoadingIndicator (/home/node/.npm/_npx/5164864a48bff686/node_modules/create-remix/dist/loading-indicator.js:46:16)
    at async loadingIndicator (/home/node/.npm/_npx/5164864a48bff686/node_modules/create-remix/dist/index.js:598:3)
    at async gitInitStep (/home/node/.npm/_npx/5164864a48bff686/node_modules/create-remix/dist/index.js:412:3)
    at async Object.createRemix (/home/node/.npm/_npx/5164864a48bff686/node_modules/create-remix/dist/index.js:77:7) {
  shortMessage: 'Command failed with exit code 128: git commit -m Initial commit from create-remix',
  command: 'git commit -m Initial commit from create-remix',
  escapedCommand: 'git commit -m "Initial commit from create-remix"',
  exitCode: 128,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
npm notice
npm notice New minor version of npm available! 10.2.4 -> 10.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.0
npm notice Run npm install -g npm@10.5.0 to update!
npm notice
```

**Comment by brophdawg11** [comment]
If you don't want the commit, you can just say "no" to the git question and run `git init`/`git commit` yourself.  For the vast majority of users, the commit succeeds and the commit locks in a known working state in the git history.
